Tests/MongoSwiftTests/APMTests.swift

```diff
@@ -0,0 +1,203 @@
+        let eventTask = Task { () -> Bool in
+            for try await _ in client.commandEventStream() {}
+            return true
+        }
+        try await client.db("admin").runCommand(["ping": 1])
+        try await client.close()
+
+        // Ensure the task returned eventually
+        _ = try await eventTask.value
+    }
+
+    func testCommandStreamHasCorrectEvents() async throws {
+        let commandStr: [String] = ["ping", "ping", "drop", "drop", "endSessions", "endSessions"]
+        let eventTypes: [EventType] = [
+            .commandStartedEvent,
+            .commandSucceededEvent,
+            .commandStartedEvent,
+            .commandSucceededEvent,
+            .commandStartedEvent,
+            .commandSucceededEvent
+        ]
+        let clientTask = try await self.withTestNamespace { client, db, _ -> Task<Int, Error> in
+            let cmdTask = Task { () -> Int in
+                var i = 0
+                for try await event in client.commandEventStream() {
+                    expect(event.commandName).to(equal(commandStr[i]))
+                    expect(event.type).to(equal(eventTypes[i]))
+                    i += 1
+                }
+                return i
+            }
+
+            try await db.runCommand(["ping": 1])
+            return cmdTask
+        }
+        let numEvents = try await clientTask.value
+        expect(numEvents).to(equal(6))
+    }
+
+    func testCommandStreamBufferingPolicy() async throws {
+        let insertsDone = NIOAtomic<Bool>.makeAtomic(value: false)
+        let clientTask = try await self.withTestNamespace { client, _, coll -> Task<Int, Error> in
+            let cmdBufferTask = Task { () -> Int in
+                var i = 0
+                let stream = client.commandEventStream()
+                try await assertIsEventuallyTrue(description: "inserts done") {
+                    insertsDone.load()
+                }
+
+                for try await _ in stream {
+                    i += 1
+                }
+                return i
+            }
+            for _ in 1...60 { // 120 events
+                try await coll.insertOne(["hello": "world"])
+            }
+            insertsDone.store(true)
+            return cmdBufferTask
+        }
+        let numEventsBuffer = try await clientTask.value
+        // Cant check for exactly 100 because of potential load balancer events getting added in during loop
+        expect(numEventsBuffer).to(beLessThan(120))
+    }
+
+    actor TaskCounter {
+        var taskCounter = 0
+
+        func incr() {
+            self.taskCounter += 1
+        }
+    }
+
+    /// Helper that tests kicking off multiple streams concurrently
+    fileprivate func concurrentStreamTestHelper<T>(f: @escaping (MongoClient) -> T) async throws
+        where T: AsyncSequence, T.Element: StreamableEvent
+    {
+        var taskArr: [Task<[EventType], Error>] = []
```

Comment:
this might be a nice place to utilize a `TaskGroup`. it's basically the same thing you have here with the array but it's a built in type. I'm not sure it would give us any extra functionality we'd actually use but it is a bit more idiomatic 

 https://developer.apple.com/documentation/swift/taskgroup

Reply:
Didnt know this existed - pretty cool. Seems to work but experimentation showed that I need to switch the order of `withTestClient` and the tasks, i.e. 
```
try await withThrowingTaskGroup( { 
    try await self.withTestNamespace{ 
     //Stream + operation code
    }
    //Collecting the outputs of the child tasks
}
//Run expectations tests
```

Also removed the actor and replaced with atomic incrementation as Patrick suggested previously